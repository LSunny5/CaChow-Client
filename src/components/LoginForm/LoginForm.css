.lForm {
    height: 100%;
    margin-top: 10vh;
    background: url(/images/LeavesBorder.jpg) no-repeat center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    font-family: 'Delius', cursive;
}

.formPage {
    height: 80vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
}


.LoginForm {
    height: 90%;
    border: 1px solid black;
    padding: 1em;
    vertical-align: middle;
    width: 90%;
    margin: 0 auto;
    border-radius: 30px;
    /* background: lightgray; */
}

.formTitle {
    font-size: 2em;
    padding-bottom: 20px;
}

.loginDirect {
    margin: 0 auto 15px;
    width: 90%;
}

.user_name, .password {
    font-weight: bold;
}

#loginUserName, #loginPassword {
    border: 1px solid #ddd;
    border-bottom: 1px solid blue;
    display: block;
    font-size: 1.1em;
    margin-bottom: 10px;
    padding: 10px;
    text-align: left;
    width: 100%;
    border-radius: 4px;
}

.loginButton, .registerButton {
    margin: 15px auto 20px;
    order: 1px solid #aaa;
    background-color: #eee;
    font-size: 1.1em;
    border-radius: 10px;
    padding: 8px 12px;
    font-weight: bold;
}

.registerDesc {
    margin-bottom: 5px;
}

.red {
    color: red;
    background: white;
    border: 1px solid black;
    padding: 10px;
    margin: 10px;
    font-size: 1.2em;
    font-weight: 600;
}

.demoInstructions {
    margin: 20px;
    border: 1px dotted black;
    padding: 10px;
    background: lightblue;
}

.makeBold {
    font-weight: bolder;
}