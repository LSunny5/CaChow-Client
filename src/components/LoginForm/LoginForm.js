import React, { Component } from 'react';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
import './LoginForm.css';
import { NavLink } from 'react-router-dom';

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({ error: null })
    const { user_name, password } = ev.target

    AuthApiService.postLogin({ 
      user_name: user_name.value.toLowerCase().trim(),
      password: password.value,
    })
      .then(res => {
        localStorage.setItem('name', user_name.value);
        user_name.value = ""; 
        password.value = "";
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    localStorage.clear();
    
    return (
      <form
        className='LoginForm'
        onSubmit={this.handleSubmitJwtAuth}
      >
        <legend className="loginDirect">Please enter your username and password. </legend>
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='user_name'>
          <label htmlFor='loginUserName'>
            Username <span className="required">*</span>
          </label>
          <input
            required
            name='user_name'
            id='loginUserName'>
          </input>
        </div>
        <div className='password'>
          <label htmlFor='loginPassword'>
            Password <span className="required">*</span>
          </label>
          <input
            required
            name='password'
            type='password'
            id='loginPassword'>
          </input>
        </div>
        <button type='submit' className="loginButton">
          Log in
        </button>
        <div className="registerQuestionBox">
          <p className="registerDesc">
            No Account? No Problem!
          </p>
          <NavLink to='/register' className="registerLink">
            Click here to Register!
          </NavLink>
        </div>
      </form>
    )
  }
}