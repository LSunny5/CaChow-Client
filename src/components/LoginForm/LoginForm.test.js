import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`LoginForm route`, () => {
    const props = {
        error: {}, 
        onLoginSuccess: () => {}, 
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <LoginForm {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the LoginForm form', () => {
        const wrapper = shallow(<LoginForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});