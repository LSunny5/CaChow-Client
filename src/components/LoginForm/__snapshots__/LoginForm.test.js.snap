// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm route renders the LoginForm form 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <legend
    className="formTitle"
  >
    Login
  </legend>
  <p
    className="loginDirect"
  >
    Please enter your username and password. 
  </p>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="loginUserName"
    >
      Username 
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      id="loginUserName"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="loginPassword"
    >
      Password 
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      id="loginPassword"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="buttonBox"
  >
    <button
      className="borderButton"
      type="submit"
    >
      Log in
    </button>
  </div>
  <div
    className="registerQuestionBox"
  >
    <p
      className="registerDesc"
    >
      No Account? No Problem!
    </p>
    <NavLink
      className="registerLink"
      to="/register"
    >
      Click here to Register!
    </NavLink>
  </div>
</form>
`;
